### PR TITLE
New Lua Scripting

### PR DESCRIPTION
**Pull request by megumumpkin**
Here's the new Lua scripting which will replace the old one.
New scripts now use SCRIPT_DIR() to get script's current path unlike before which is automatic.
Also new functionality for users to have more control over their script (Killing exactly 1 coroutine, or 1 instance, or instances related by 1 file), also for script reloads there are new functionality to detect if script has been run once before.

### CODE REVIEW

**Review by turanszkij — CHANGES_REQUESTED** [review]
- Sample scripts do not work: 
image

- I'd like if the `SCRIPT_DIR()` function was renamed to `script_dir()` instead, and same for `script_file()`, `script_pid()`. `Script_Initialized()` to `script_initialized()`

File: WickedEngine/wiLua.cpp [hunk 1]
```diff
@@ -42,25 +42,64 @@ namespace wi::lua
...
+				_ENV.PROCESSES_DATA[SCRIPT_PID()]._INITIALIZED = _ENV.PROCESSES_DATA[SCRIPT_PID()]._INITIALIZED + 1
+			end
+		)";
+
+		if(PID == 0){
+			PID = Internal_GenScriptPID();
+		}
+
+		std::string dynamic_inject = "local function SCRIPT_FILE() return \""+filename+"\" end\n";
+		dynamic_inject += "local function SCRIPT_PID() return \""+std::to_string(PID)+"\" end\n";
```
Comment (turanszkij) [inline WickedEngine/wiLua.cpp:70]: Why after `return there is a `\` character? I think this causes error on Windows, I don't think this is needed.

Comment (turanszkij) [reply WickedEngine/wiLua.cpp:70]: Oh, it's for the `"`, I see

**Comment by turanszkij** [comment]
Oh, I think the error is the path itself, because it contains `\` which lua recognizes as escape char..

**Review by turanszkij — APPROVED** [review]

**Comment by turanszkij** [comment]
Done fixes and some changes, let me know if you find any issues or ok to merge.

**Comment by megumumpkin** [comment]
Ok to merge, no issues

**Pull request merged**